pkg/traceql/storage.go

```diff
@@ -84,6 +84,10 @@ type Span interface {
 	ID() []byte
 	StartTimeUnixNanos() uint64
 	DurationNanos() uint64
+
+	SiblingOf(Span) bool
```

Comment:
is there a way to work this into being returned through the existing `Attributes()` func? maybe 
```
span.Attributes()[NewIntrinsic(IntrinsicStructuralDescendant)]
```
returns a func that can be executed to get the desired results? i find these a little awkward but i can think of no good alternatives. i'm afraid this pattern will grow and i'm trying to think of a way to jam it into the existing attributes. wdyt?

- returning left/right/parent binds the engine to the current implementation
- returning slices of valid siblings/descendants/children is far more costly then what you've done

Reply:
What do you see are the pros of using Attributes()?  I agree that this interface is feeling unfinished and the split between dedicated fields (like ID()) and dynamic Attributes() isn't well-defined.  Maybe one idea is that anything addressable by a native TraceQL enum/attr (i.e. IntrinsicName) could go in Attributes(), and everything else has a dedicated field.  Like Span.ID() could be eliminated and referenced by `span.Attributes()[NewIntrinsic(IntrinsicSpanID)]` now.

To bring it back to the original question:  if we exposed the parent/left/right values to the engine and bound the implementation, we'd switch to real IntrinsicLeft/Right/Parent and put them in Attributes.  However I really think the current interface is a better abstraction but we won't know it until we implement TraceQL on another format (I really want to do things like TraceQL tail-sampling, compaction rules, etc).

Reply:
> What do you see are the pros of using Attributes()?

Mainly the consistency of all data requested from the Fetch layer coming through the same path. Also simplicity of the interface is nice. It's a good point that intrinsics like span id and duration are coming through predefined functions. That did not occur to me while making the comment.

> To bring it back to the original question: if we exposed the parent/left/right values to the engine and bound the implementation, we'd switch to real IntrinsicLeft/Right/Parent and put them in Attributes. However I really think the current interface is a better abstraction but we won't know it until we implement TraceQL on another format (I really want to do things like TraceQL tail-sampling, compaction rules, etc).

Agree the current implementation is superior to adding left/right/parent or all valid relationship data in slices. This is definitely not a blocker. I just wanted to muse on it some since it is expanding the interface.